flink-connectors/flink-hbase/src/main/java/org/apache/flink/addons/hbase/util/HBaseTypeUtils.java

```diff
@@ -81,13 +81,16 @@ public static Object deserializeToObject(byte[] value, int typeIdx, Charset stri
 	 * Serialize the Java Object to byte array with the given type.
 	 */
 	public static byte[] serializeFromObject(Object value, int typeIdx, Charset stringCharset) {
+		if (value == null){
+			return EMPTY_BYTES;
```

Comment:
Could you add a dual logic in the `deserializeToObject`? If the input value is EMPTY_BYTES, we can just return null. I guess we will have similar problem in the deserialization side?

Reply:
@wuchong ,hi , i check the code, when deserializeToObject called, there could be a same problem. But when i dig deeper, It seems ok. 

There're two place where `deserializeToObject` is called.
One(I think it's ok) is like the following in `HBaseReadWriteHelper.parseToRow(Result result, Object rowKey`):
```java
    if (value != null) {					
          familyRow.setField(q, HBaseTypeUtils.deserializeToObject(value, typeIdx, charset));
     } else {
	familyRow.setField(q, null);
    }
```

The other one also looks good (in `HBaseReadWriteHelper.parseToRow(Result result)`) : 

```java 
public Row parseToRow(Result result) {
		if (rowKeyIndex == -1) {
			return parseToRow(result, null);
		} else {
			Object rowkey = HBaseTypeUtils.deserializeToObject(result.getRow(), rowKeyType, charset);
			return parseToRow(result, rowkey);
		}
	}
```

I can't think of a scenario that result.getRow could return a null value because this code is used to parse the returned value form hbase with a scan command. 
Do you have any suggestion?

Reply:
The rowkey from HBase will never be null or emtpy bytes, but the qualifiers will be an empty bytes because that's what we write in the `serializeFromObject`.

Reply:
Yeah, i get it. f there's an emtpy value sent to deserializeToObject, It's not a NPE exception.

Do you think we should handle the exception in this PR? 
 If not, i could create another issue to track the exception in deserializeToObject.

Reply:
Yes, I think we should better handle in this PR, because the empty bytes (for types exclude bytes and string) are introduced in this PR.

Reply:
@wuchong Hi,  handle the empty bytes in `deserializeToObject` under your suggestion. Please help review. Many thanks.

Reply:
@wuchong , Hi, when i handle the empty byte by returning a null value, Some tests go wrong, like 'testTableSourceSinkWithDDL','testTableSink' . Because they expect an empty string when the value is empty bytes.  Here we have two options :

1. **Only return a null value when the type is not string (typeIdx ==1), for string type , return an empty string "".**
```java
if (Arrays.equals(value, EMPTY_BYTES)){
	if (typeIdx == 1){
		return "";
	}else{
		return null;
	}
}
```
2. **modifying the test case; making the expect null other than empty string when the value of an column in hbase is empty bytes**

I think we'd better modifying the test case, because when flink writes a null to hbase, we should read it from hbase as null. Even for an existing hbase table, if the value of a column is empty bytes , we take it  as null .  It also works in the same logic. 

How do you think about it?

Reply:
We introduced a `null-string-literal` in the new hbase connector (see `HBaseDynamicTableFactory`) to handle this problem. I think we can introduce a same option in the old hbase connector.